src/server_interface/server_interface.js

```diff
@@ -289,26 +290,24 @@ var CardshifterServerAPI = {
     },
 
     /**
-    * Sets an event listener for when the server sends a message and
-    * the message type is one of the types in types
+    * Sets an event listener for when the server sends a message
+    * and the message command is one of the keys in the commandMap
     *
-    * @param listener:Function -- The function to fire when a message of types is received
-    * @param types:[string] (OPTIONAL) -- Only fire the listener when the message type is in this array
-    * @param timeout:Object (OPTIONAL) -- The function(.ontimeout) to call after MS(.ms) of no reply
-    *
-    * TODO: Maybe a timeout will be needed? Pass in a function and a MS count.
+    * @param commandMap:Object -- The keys are the command types and the values are
+    *       functions to run when a message of that command is encountered.
+    * @param scope:$scope -- The scope of the controller running this.
     */
-    setMessageListener: function(listener, types) {
-        eventTypes = types;
-
+    setMessageListener: function(commandMap, scope) {
         this.socket.onmessage = function(message) {
             var data = JSON.parse(message.data);
-            if(eventTypes) {
-                if(eventTypes.indexOf(data.command) !== -1) { // if contains
-                    listener(data);
-                }
-            } else {
-                listener(data);
+            var command = data.command;
+
+            console.log("Received message", data);
```

Comment:
Hmm, okay. Then perhaps you could look at using [debug](https://www.npmjs.com/package/debug)?